From 64f9913631cdca7b348cbbc272ea2d4d0c944ccd Mon Sep 17 00:00:00 2001
From: Mathieu Duponchelle <mathieu@centricular.com>
Date: Fri, 18 Jun 2021 19:26:35 +0200
Subject: [PATCH] x265enc: add negative DTS support

Use the same set_min_pts approach as x264enc.

Fixes: https://gitlab.freedesktop.org/gstreamer/gst-plugins-bad/-/issues/304
Part-of: <https://gitlab.freedesktop.org/gstreamer/gst-plugins-bad/-/merge_requests/2340>
---
 ext/x265/gstx265enc.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/ext/x265/gstx265enc.c b/ext/x265/gstx265enc.c
index d0dc3f761..93d8ed384 100644
--- a/ext/x265/gstx265enc.c
+++ b/ext/x265/gstx265enc.c
@@ -641,6 +641,10 @@ gst_x265_enc_start (GstVideoEncoder * encoder)
 
   g_ptr_array_set_size (x265enc->peer_profiles, 0);
 
+  /* make sure that we have enough time for first DTS,
+     this is probably overkill for most streams */
+  gst_video_encoder_set_min_pts (encoder, GST_SECOND * 60 * 60 * 1000);
+
   return TRUE;
 }
 
-- 
2.32.0

